### PR TITLE
fix: add support for Elixir v1.17.0

### PR DESCRIPTION
**Pull request by sheerlox**
Elixir v1.17.0 just released. The install script is failing because the OTP version if/else block isn't up-to-date.

Renovate is effectively broken for Elixir right now: https://github.com/sheerlox/elixir_renovate_demo/pull/6

Renovate log from the PR above (Mend Renovate job id `7e88e954-625d-4a07-804b-45eefc1a151c`):
```
[17:27:45.046] INFO (193): Installing tool elixir@1.17.0...
installing v2 tool elixir v1.17.0
Download failed: https://github.com/elixir-lang/elixir/releases/download/v1.17.0/elixir-otp-24.zip
[17:27:45.616] INFO (221): Downloading file ...
    url: "https://github.com/elixir-lang/elixir/releases/download/v1.17.0/elixir-otp-24.zip"
    output: "/tmp/renovate/cache/containerbase/f7649696d9e36012fd17fe14a3344673897f8c089328bc8a15abaf579e0f5ab1/elixir-otp-24.zip"
[17:27:45.764] FATAL (221): Response code 404 (Not Found)
    err: {
      "type": "HTTPError",
      "message": "Response code 404 (Not Found)",
      // ...
    }
[17:27:45.890] INFO (221): Download completed with errors  in 281ms.
[17:27:45.901] FATAL (193): Command failed with exit code 1: /usr/local/containerbase/bin/install-tool.sh elixir 1.17.0
    err: {
      "type": "Error",
      "message": "Command failed with exit code 1: /usr/local/containerbase/bin/install-tool.sh elixir 1.17.0",
      "stack":
          Error: Command failed with exit code 1: /usr/local/containerbase/bin/install-tool.sh elixir 1.17.0
              at makeError (/snapshot/dist/containerbase-cli.js:39455:13)
              at handlePromise (/snapshot/dist/containerbase-cli.js:40170:29)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async InstallLegacyToolService.execute (/snapshot/dist/containerbase-cli.js:56573:5)
              at async InstallToolService.execute (/snapshot/dist/containerbase-cli.js:56765:9)
              at async InstallToolShortCommand.execute (/snapshot/dist/containerbase-cli.js:57006:14)
              at async InstallToolShortCommand.validateAndExecute (/snapshot/dist/containerbase-cli.js:2428:26)
              at async _Cli.run (/snapshot/dist/containerbase-cli.js:3541:22)
              at async _Cli.runExit (/snapshot/dist/containerbase-cli.js:3549:28)
              at async main (/snapshot/dist/containerbase-cli.js:57200:3)
      "shortMessage": "Command failed with exit code 1: /usr/local/containerbase/bin/install-tool.sh elixir 1.17.0",
      "command": "/usr/local/containerbase/bin/install-tool.sh elixir 1.17.0",
      "escapedCommand": "\\"/usr/local/containerbase/bin/install-tool.sh\\" elixir 1.17.0",
      "exitCode": 1,
      "cwd": "/tmp/renovate/repos/github/sheerlox/elixir_renovate_demo",
      "failed": true,
      "timedOut": false,
      "isCanceled": false,
      "killed": false
    }
[17:27:46.150] INFO (193): Installed tool elixir with errors in 1.1s.
```

### CODE REVIEW

**Comment by rarkins** [comment]
Thanks for the PR. I was wondering: should we pick the minimum supported, the maximum, or try to find a common It looks like 1.14-1.17 all support 25, for example

**Comment by sheerlox** [comment]
Are there any benefits for this container image to use a common like 25?

The reason I've picked the latest for `1.17` is so we don't need to upgrade this script back for longer :sweat_smile: 

For `1.14` I just kept what was already in there.

**Comment by sheerlox** [comment]
Didn't realize the script supported versions < 1.14, so initially got a bit lazy with my bash operators lol. That should fix it.

**Comment by sheerlox** [comment]
> Are there any benefits for this container image to use a common like 25?

I guess the benefit is simplicity, I've updated so 25 is used from 1.14 to 1.16 (keeping 27 for 1.17 for the reason exposed above)

**Review by viceice — CHANGES_REQUESTED** [review]
don't remove compability code for older elixir versions

**Comment by viceice** [comment]
because someone can rely on older erlang

**Comment by sheerlox** [comment]
> because someone can rely on older erlang

If Erlang can be installed alongside Elixir, then this script should probably be aware of which Erlang version is installed right?

**Comment by sheerlox** [comment]
> don't remove compability code for older elixir versions

Although not expressed in the same way, my latest commit should now preserve the previous behavior

File: src/usr/local/containerbase/tools/v2/elixir.sh [hunk 1]
```diff
@@ -8,12 +8,14 @@ function install_tool () {
...
   check_command erl
 
-
   # https://github.com/elixir-lang/elixir/releases/tag/v1.14.0
-  if [ "$MAJOR" -eq 1 ] && [ "$MINOR" -eq 14 ]; then
-    base_file=elixir-otp-23.zip
-  elif [ "$MAJOR" -eq 1 ] && [ "$MINOR" -gt 14 ]; then
+  # https://hexdocs.pm/elixir/compatibility-and-deprecations.html#between-elixir-and-erlang-otp
+  if [ "$MAJOR" -eq 1 ] && [ "$MINOR" -ge 17 ]; then
+    base_file=elixir-otp-27.zip
```
Comment (viceice) [inline src/usr/local/containerbase/tools/v2/elixir.sh:14]: can't we use the lowest version instead, so it's compatible with more erlang major versions? I'll   refactor this code soon to use the current installed erlang major as base and install the most appropriate elixir version.

Comment (sheerlox) [reply src/usr/local/containerbase/tools/v2/elixir.sh:14]: You're right. I was misunderstanding Elixir builds by thinking they were compiled for a specific OTP version and therefore were incompatible with any other OTP version.

- https://elixir-lang.org/install.html#precompiled-package
- https://github.com/hexpm/bob?tab=readme-ov-file#elixir-builds

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
@sheerlox thanks a lot for the fix!

**Comment by sheerlox** [comment]
Looks like [the release workflow failed](https://github.com/containerbase/base/actions/runs/9502465307/job/26191884594) because ppa.launchpad.net was unavailable (`503  Service Unavailable`).

Any chance we can retry the failed job? :pray:

**Comment by viceice** [comment]
released around two hours ago 🤗

**Comment by sheerlox** [comment]
Can confirm Elixir dependencies upgrade is working again in Renovate `37.407.4` (on a private GitHub Actions workflow).

Mend Renovate is still behind though so FOSS repositories are still impacted.

image

**Comment by rarkins** [comment]
@nabeelsaabna can you update the app on Sunday?

**Pull request merged**